zk_toolbox/crates/zk_inception/src/commands/contract_verifier/args/init.rs

```diff
@@ -0,0 +1,155 @@
+use std::str::FromStr;
+
+use anyhow::Context;
+use clap::{Parser, ValueEnum};
+use common::{cmd::Cmd, spinner::Spinner, PromptSelect};
+use strum::IntoEnumIterator;
+use xshell::{cmd, Shell};
+
+use crate::messages::{
+    MSG_ARCH_SLECTION_PROMPT, MSG_FETCHING_ZKSOLC_RELEASES_SPINNER, MSG_GET_ZKSOLC_RELEASES_ERR,
+    MSG_INVALID_ARCH_ERR, MSG_NO_RELEASES_FOUND_ERR, MSG_NO_VERSION_FOUND_ERR,
+    MSG_ZKSOLC_VERSION_PROMPT,
+};
+
+#[derive(Debug, Clone, Parser, Default)]
+pub struct InitContractVerifierArgs {
+    /// Version of zksolc to install
+    #[clap(long)]
+    pub version: Option<String>,
+    /// Architecture of zksolc to install
+    #[clap(long)]
+    pub arch: Option<Arch>,
+}
+
+#[derive(Debug, Clone)]
+pub struct InitContractVerifierArgsFinal {
+    pub zksolc_version: ZkSolcVersion,
+}
+
+#[derive(Debug, Clone)]
+pub struct ZkSolcVersion {
+    pub version: String,
+    pub arch: Arch,
+    pub url: String,
+}
+
+#[derive(Debug, Clone, ValueEnum, strum::EnumIter, PartialEq, Eq, Copy)]
+pub enum Arch {
+    LinuxAmd,
+    LinuxArm,
+    MacosAmd,
+    MacosArm,
+    WindowsAmd,
+}
```

Comment:
should we simply use https://doc.rust-lang.org/std/env/consts/constant.OS.html and https://doc.rust-lang.org/std/env/consts/constant.ARCH.html